System/Directory/OsPath.hs

```diff
@@ -751,7 +752,13 @@ withReplacementFile :: OsPath            -- ^ Destination file
 withReplacementFile path postAction action =
   (`ioeAddLocation` "withReplacementFile") `modifyIOError` do
     mask $ \ restore -> do
-      (tmpFPath, hTmp) <- OS.openBinaryTempFile (takeDirectory path) [osstr|.copyFile.tmp|]
+      let tmpPath = case encodeWith (mkUTF8 ErrorOnCodingFailure)
+                                    (mkUTF16le ErrorOnCodingFailure)
+                                    ".copyFile.tmp"
+                         of
+            Left err -> error ("withReplacementFile: invalid encoding: " ++ show err)
```

Comment:
`error` is a bit depressing, why don't we throw an IO error?

Reply:
It's not something that should really happen under any usage of the library, so I don't see the benefit of `throwIO` which makes it easier to catch.